service/src/main/java/bio/terra/workspace/service/resource/ResourceValidationUtils.java

```diff
@@ -489,4 +493,22 @@ public static void validateCloningInstructions(
               cloningInstructions.toString(), stewardshipType.toString()));
     }
   }
+
+  /**
+   * Validate the terra reserved properties has valid values.
+   *
+   * @param properties of a resource.
+   */
+  public static void validateProperties(Map<String, String> properties) {
+    if (properties.containsKey(FOLDER_ID_KEY)) {
+      try {
+        var uuid = UUID.fromString(properties.get(FOLDER_ID_KEY));
+      } catch (IllegalArgumentException e) {
+        throw new BadRequestException(
+            String.format(
+                "Properties contains an invalid non-UUID format folder id %s.",
+                properties.get(FOLDER_ID_KEY)));
+      }
+    }
```

Comment:
we are doing that already for parent folder id.
https://cs.github.com/DataBiosphere/terra-workspace-manager/blob/7f9cd647850acb6786f57e1e7a20b4af3fe57e86/service/src/main/java/bio/terra/workspace/db/FolderDao.java?q=folderdao#L69
and 
https://cs.github.com/DataBiosphere/terra-workspace-manager/blob/7f9cd647850acb6786f57e1e7a20b4af3fe57e86/service/src/main/java/bio/terra/workspace/db/FolderDao.java?q=folderdao#L130

Reply:
ah i remember why i can't. 

i can't pass in ResourceValidationUtils into WsmResource.

Reply:
I see... I don't see a clean alternative. For updating resource, there's just one controller method. But for creating resources, we'd have to change every create method.

I think this is a Dan question. I can ask him after he gets back. Or feel free to ask him. Up to you if you want to block on this or not.

Reply:
UI is currently treating resource as a top-level resource when the folder id doesn't exist. So iMO i prefer the simplicity.

Reply:
This is actually causing us some pain because on folder delete the resources briefly jump to the root.

Reply:
that's different issue (maybe address by concurrency)

this is only in the resource creation.

Reply:
Sounds like two interesting questions here.
Do we have a ticket for the "jumping to root" problem?

Reply:
i'm puzzled how "jump to root" happens.

we delete resource before we delete folders. cloud resource deletion can be async but the metadata in the database should be deleted before we got to the folder?

which bring up to a third question :) why should cloud resource deletion async-able in a stairway flight which is synchronously executed every step? https://cs.github.com/DataBiosphere/terra-workspace-manager/blob/7f9cd647850acb6786f57e1e7a20b4af3fe57e86/service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/delete/DeleteControlledResourcesFlight.java?q=controlledresources+delete+flight#L60

Reply:
"jump to root" is a combination of two things:
- a race between when we refresh the folder list and refresh the resource list.  so even if both are deleted we may get the new folder list before the new resource list, so the resource still exists with an invalid folder.
- we made the UI show resources without a valid folder as appearing in the root.  this is probably a bad idea and we should revert it.  especially if this PR makes it such that this can't happen in the BE

Reply:
the race should be addressed separately.

maybe when the folder is being deleted, the folder and resources under will be marked as deleting and you just hide the list  ( i think that's what the product want?) or not refresh

@cbookg or @ddietterich  mind take another look at the PR?